### DISCUSSION TITLE
Just a thought: custom css class ?

### DISCUSSION THREAD

**Discussion by Moyf** [chrisgurney/obsidian-note-toolbar#103] [Ideas]
Just wanted to set some custom styles for a specific Toolbar, and thinks that there is a sort-of style settings here:
![image](https://github.com/user-attachments/assets/90c604e1-ac05-4c48-b7db-915e5407f981)

What if adding a input box for  `custom css`, allowing users to apply css class to a specific note toolbar ?

But the `cssclasses` property of note might also serve a similar purpose... The difference is that the css in NTB setting can be inherited across different pages.

Anyway, this is just a casual thought, and I'm not entirely sure is it necessary  :D

**Comment by chrisgurney** [comment]
@Moyf Very interesting! Let me think on this one.

**Reply by chrisgurney** [reply]
## Initial thoughts

This would make for a very flexible solution to several problems I can't think of at the moment, aside from advanced vault styling. 😄 That said:

- I don't think this is a lot of effort to add.
- The bulk of the work is in the settings UI: On clicking a `custom styles` / `custom classes` / `custom CSS` option added to that list, a setting field would appear below, allowing for entering one or more classes (in a space-delimited list).
- I'd be tempted to also allow for styles (in a semi-colon delimited list) to be directly added _in addition to_ classes, so that it's easier for less technical people to see and copy/paste styling solutions from other users.

I think I'd like to see some use cases from multiple users where this would be a viable solution, before I go ahead and give this a go!

**Comment by stracaganasse** [comment]
For reference this was the first feature i searched for when i did not find the `space-between` flex option equivalent for elements spacing. 
Could still be a simple PR to add it for my specific usecase though, still unsure of other needs

**Reply by chrisgurney** [reply]
I might go ahead and add that option to Styles. Let me give it a go.

If I add it, it will be part of the beta at first. 

You could override the "even" style (if you're not using it), until the new style is released.

**Reply by chrisgurney** [reply]
@stracaganasse Great suggestion. TBH, I'm not sure why I didn't add it in the first place.

The new `between` style will be available in the next beta. [Here's the commit.](https://github.com/chrisgurney/obsidian-note-toolbar/commit/74a36ae477e2883e19671abafa5063ea15b89e06)

**Comment by chrisgurney** [comment]
@Moyf You can now add your own custom classes, as defined in CSS snippets, to specific toolbars. See [`1.16-beta-08`](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.16-beta-08) and [the documentation](https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Custom-styling). 

Let me know what you think of the implementation, and thanks for the idea!

**Reply by Moyf** [reply]
That's great! Thank you! 🥳
